lib/std/multi_array_list.zig

```diff
@@ -174,10 +177,28 @@ pub fn MultiArrayList(comptime T: type) type {
 
         /// Release all allocated memory.
         pub fn deinit(self: *Self, gpa: Allocator) void {
+            self.pointer_stability.assertUnlocked();
             gpa.free(self.allocatedBytes());
             self.* = undefined;
         }
 
+        /// Puts the MultiArrayList into a state where any method call that
+        /// would cause an existing item pointer to become invalidated will
+        /// instead trigger an assertion.
+        ///
+        /// An additional call to `lockPointers` in such state also triggers an
+        /// assertion.
+        ///
+        /// `unlockPointers` returns the MultiArrayList to the previous state.
+        pub fn lockPointers(self: *Self) void {
+            self.pointer_stability.lock();
+        }
+
+        /// Undoes a call to `lockPointers`.
+        pub fn unlockPointers(self: *Self) void {
+            self.pointer_stability.unlock();
+        }
+
         /// The caller owns the returned memory. Empties this MultiArrayList.
         pub fn toOwnedSlice(self: *Self) Slice {
```

Comment:
I'm not sure if we should be asserting unlocked when we convert to an owned slice.

Reply:
please see the `move` method of ArrayHashMap for inspiration